src/sys/unix/awakener.rs

```diff
@@ -72,3 +77,66 @@ mod pipe {
         }
     }
 }
+
+/// Awakener backed by an eventfd on Linux
+#[cfg(any(target_os = "linux", target_os = "android"))]
+mod eventfd {
+    use sys::unix;
+    use {io, Ready, Poll, PollOpt, Token};
+    use event::Evented;
+    use std::io::{Read, Write};
+    use std::mem::transmute;
+
+    /*
+     *
+     * ===== Awakener =====
+     *
+     */
+
+    pub struct Awakener {
+        fd: unix::Io,
+    }
+
+    impl Awakener {
+        pub fn new() -> io::Result<Awakener> {
+            Ok(Awakener { fd: unix::eventfd()? })
+        }
+
+        pub fn wakeup(&self) -> io::Result<()> {
+            let data: [u8; 8] = unsafe { transmute(1u64) };
+            match (&self.fd).write(&data) {
+                Ok(_) => Ok(()),
+                Err(e) => {
+                    if e.kind() == io::ErrorKind::WouldBlock {
```

Comment:
I don't think hiding this error is a good idea. If this return a `WouldBlock` error it means that wake is called 2^64 times, without it being clean up. This is an user error in my opinion. Furthermore the pol instance won't actually be woken, another reason to return an error to let the user know.

Reply:
`WouldBlock` cannot be called 2^64 times.  Even if it could be called once per clock cycle, it would still take more than a century.

Reply:
@asomers right, so why bother with checking for the error in the first place?

Reply:
For "completeness". Even though it is unthinkable that the counter can overflow, if it ever did, this is how it should be handled (the same way it is handled a full buffer in the pipe awakener). I just like be aware of corner cases, but feel free to remove the check.